pkg/backend/filestate/backend_test.go

```diff
@@ -466,3 +467,27 @@ func TestHtmlEscaping(t *testing.T) {
 	state := string(bytes)
 	assert.Contains(t, state, "<html@tags>")
 }
+
+func TestLocalBackendRejectsStackInitOptions(t *testing.T) {
+	t.Parallel()
+
+	// • Create an empty struct to simulate a non-nil value.
+	type stackInitOptions struct{}
+	var opts = &stackInitOptions{}
+
+	// • Create a mock local backend
+	var tmpDir = t.TempDir()
+	var dirURI = fmt.Sprintf("file://%s", filepath.ToSlash(tmpDir))
+	var local, err = New(cmdutil.Diag(), dirURI)
+	assert.NoError(t, err)
+	var ctx = context.Background()
+
+	// • Simulate `pulumi stack init`, passing non-nil init options
+	fakeStackRef, err := local.ParseStackReference("foobar")
+	assert.NoError(t, err)
+	assert.Panics(t, func() {
+		// • Expect an error.
+		_, err := local.CreateStack(ctx, fakeStackRef, opts)
```

Comment:
Nothing uses this "opts" parameter yet. Maybe just kill it and pass a typed structure instead with "Teams []string" then make localbackend error (not panic) on teams being non empty.

Reply:
If I'm understanding this right, this suggestion will eliminate the `interface{}` used to plumb options to backends. I'm in favor.

That'll also get rid of the SupportsTeams method because the backend can just inspect the struct and decide if it supports teams.

Reply:
Do you mean reflect on the struct type to see if it supports teams or not?

Reply:
I read this as introducing a  `backend.CreateStackOptions` struct to replace the `opts interface{}` in the `Backend` interface so that we don't need to do this interface matching at all.

```
package backend

type CreateStackOptions struct {
  Teams []string
}

type Backend interface {
  // ...

  CreateStack(
	ctx context.Context, stackRef backend.StackReference, opts CreateStackOptions) (backend.Stack, error)
}
```

Reply:
Yes. I think we're largely in agreement about the benefits of removing `interface{}`. Just want to offer one point of disagreement and see what yinz think.

However, Teams are only supported by the HTTP backend. So this field would be zero'd in other implementers of `Backend`. My issue is that it's a common footgun in Go to have zeroed, uninitialized state that's indistinguishable from initialized state. I'm leaning on the "make illegal states unrepresentable" philosophy here. In a functional language, we could distinguish between `Option<[]string>` and `[]string` for this case, where `Some([]string{})` represents provided but empty array (useful for the HTTP backend) and `None` represents "Not supported". In Go, we can't really implement this pattern (some gophers use `*[]string` in this case but that's asking for a null-pointer deref.)

I'd like to lean on the "[accept interfaces and return structs](https://subscription.packtpub.com/book/programming/9781838647940/7/ch07lvl1sec45/accepting-interfaces-and-returning-structs)" methodology here:

```golang
package backend

type CreateStackOptions interface {
  SupportsTeams() bool
  Teams() []string
}

type Backend interface {
  // ...

  CreateStack(
	ctx context.Context, stackRef backend.StackReference, opts CreateStackOptions) (backend.Stack, error)
}
```

which I think is idiomatic.

Reply:
Oops, let me change that signature:

```golang
type CreateStackOptions interface {
  Teams() ([]string, error)
}

type TeamsNotSupportedError string
```

Reply:
I've resolved this discussion because I introduced a `CreateStackOptions` interface as per the discussion.